utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/encode/AvroSchemaOutputValidator.scala

```diff
@@ -0,0 +1,236 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+import cats.data.Validated.condNel
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.avro.Schema.Type
+import org.apache.avro.SchemaCompatibility.SchemaCompatibilityType
+import org.apache.avro.{LogicalTypes, Schema, SchemaCompatibility}
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.process._
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.avro.AvroUtils
+import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
+
+import java.nio.ByteBuffer
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+private[encode] object AvroSchemaExpected {
+
+  private val NullSchema = Schema.create(Schema.Type.NULL)
+
+  def apply(schema: Schema, isNullable: Boolean): AvroSchemaExpected =  {
+    val expectedSchema = if (isNullable) {
+      Schema.createUnion(NullSchema, schema)
+    } else {
+      schema
+    }
+
+    AvroSchemaExpected(expectedSchema)
+  }
+
+}
+
+private[encode] case class AvroSchemaExpected(schema: Schema) extends OutputValidatorExpected {
+  override def expected: String = AvroSchemaOutputValidatorPrinter.print(schema)
+}
+
+object AvroSchemaOutputValidator {
+  private[encode] val SimpleAvroPath = "Data"
+}
+
+class AvroSchemaOutputValidator(validationMode: ValidationMode) extends LazyLogging {
+
+  import AvroSchemaOutputValidator._
+
+  import scala.collection.JavaConverters._
+
+  private val nestedObjects = Set(Type.RECORD, Type.MAP, Type.ARRAY)
+
+  private val valid = Validated.Valid(())
+
+  private val longLogicalTypes = Set(
+    LogicalTypes.timeMicros(), LogicalTypes.timestampMillis(), LogicalTypes.timestampMicros()
+  )
+
+  /**
+    * see {@link pl.touk.nussknacker.engine.avro.encode.BestEffortAvroEncoder} for underlying avro types
+    */
+  def validateTypingResultToSchema(typingResult: TypingResult, parentSchema: Schema)(implicit nodeId: NodeId): ValidatedNel[OutputValidatorError, Unit] =
+    validateTypingResult(typingResult, parentSchema, None)(isNullable = false)
+
+  /**
+    * isNullable is kind of workaround... In nullable nested schemas ([null, Record] / [null, Map] / [null, List]) we
```

Comment:
I assume we want to present information about schema in best way.. So, please imagine situation when you have union: [null, List[Any]] and when someone put there just Integer or other type then will be nice so information about that field type is null | List[Any]. 

This is use only when we have nullable union, because when we do comparing inside object then we want to see information only about comparing elements, we don't want to see information about general nullability of root object -> it's difficult to explain, but maybe we can just talk about it?

Reply:
After discussion we decided to remove it.